pennylane/hf/tapering.py

```diff
@@ -526,3 +538,77 @@ def optimal_sector(qubit_op, generators, active_electrons):
         perm.append(coeff)
 
     return perm
+
+
+def transform_hf(generators, paulix_ops, paulix_sector, num_electrons, num_wires):
+    r"""Transform a Hartree-Fock state with a Clifford operator and taper qubits.
+
+    The Hartree-Fock (HF) state for a molecule is transformed to a qubit observable under Jordan-Wigner (JW)
+    transform. To do this, each occupied orbital in the original HF state is assigned with a fermionic creation
+    operator and each unoccupied orbital is replaced with an identity operator. The JW transform is then applied to
+    map the resulting fermionic operator to a qubit basis.
+
+    This observable is tapered using the same Cliffords :math:`U` that are obtained from the symmetries of the
+    molecular Hamiltonian. A new, tapered Hartree-Fock state is built by placing all the qubits (wires) which are
+    acted on by a Pauli-X or Pauli-Y operator in the :math:`|1\rangle` state, while leaving the rest of them in
+    the :math:`|0\rangle` state.
+
+    Args:
+        generators (list[Hamiltonian]): list of generators of symmetries, taus, for the Hamiltonian
+        paulix_ops (list[Operation]):  list of single-qubit Pauli X operators
+        paulix_sector (list[int]): list of eigenvalues of Pauli-X operators
+        num_electrons (int): number of active electrons in the system for generating the Hartree-Fock bitstring
+        num_wires (int): number of wires in the system for generating the Hartree-Fock bitstring
+
+    Returns:
+        array(int): tapered Hartree-Fock state
+
+    **Example**
+
+    >>> symbols = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, -0.69440367], [0.0, 0.0, 0.69440367]])
+    >>> mol = qml.hf.Molecule(symbols, geometry)
+    >>> H = qml.hf.generate_hamiltonian(mol)(geometry)
+    >>> generators, paulix_ops = qml.hf.generate_symmetries(H, len(H.wires))
+    >>> paulix_sector = qml.hf.optimal_sector(H, generators, mol.n_electrons)
+    >>> qml.hf.transform_hf(generators, pauli_x_ops, paulix_sector,
+                            mol.n_electrons, len(H.wires))
+        [1]
+    """
+
+    pauli_map = {"I": qml.Identity, "X": qml.PauliX, "Y": qml.PauliY, "Z": qml.PauliZ}
+
+    # build the untapered Hartree Fock state
```

Comment:
Hmmm, is this always correct? I guess so. Makes me wonder how `qml.qchem.hf_state()` does it, since this seems very compact